lib/libc/mips/string/Makefile.inc

```diff
@@ -12,28 +12,11 @@ MDSRCS+= \
 	strchr.S \
 	strcmp.S \
 	strrchr.S
-.endif
 
 .if ${MACHINE_CPU:Mcheri}
-.if !${MACHINE_ABI:Mpurecap}
-CHERI_MDSRCS=	\
-		memcpy_c.c \
-		memmove_c.c \
-		memset_c.c
-.endif
-
-CHERI_MISRCS=	\
-		bcopy.c \
-		memcpy.c \
-		memmove.c \
-
-MDSRCS+=	${CHERI_MDSRCS}
-
-.if ! ${MACHINE_ABI:Mpurecap}
-.for src in ${CHERI_MDSRCS} ${CHERI_MISRCS}
-CC.${src}=	${CC} -target mips64-unknown-freebsd -cheri
-.endfor
+MDSRCS+= \
+	memcpy_c.c \
+	memmove_c.c \
+	memset_c.c
```

Comment:
I think these can go entirely since they are already in MISRCS. All this does is cause extra work in this bit of `lib/libc/Makefile`:
```
# If there are no machine dependent sources, append all the
# machine-independent sources:
.if empty(MDSRCS)
SRCS+=  ${MISRCS}
.else 
# Append machine-dependent sources, then append machine-independent sources
# for which there is no machine-dependent variant.
SRCS+=  ${MDSRCS}
.for _src in ${MISRCS}
.if ${MDSRCS:R:M${_src:R}} == ""
SRCS+=  ${_src}
.endif
.endfor
.endif
```

Reply:
Actually finding the files just uses `.PATH`

Reply:
Well, there's always the risk that the file extensions will be different, but probably not for C files, only really applies for MD .[sS] files?

Reply:
Yeah, the logic is to handle that case, for .c files we can just pick up the MD .c version by path magic (and that seems to generally be the convention.)

Reply:
Ah and we already rely on that to make memcpy.c etc use jdw57_memcpy.c, so completely sensible to make memcpy_c.c do the same.